frontend-api/src/test/java/uk/gov/di/authentication/frontendapi/services/MfaResetIPVAuthorizationServiceTest.java

```diff
@@ -0,0 +1,239 @@
+        when(jwtService.signJWT(any(), any(), any())).thenReturn(TEST_SIGNED_JWT);
+        when(jwtService.encryptJWT(any(), any())).thenReturn(TEST_ENCRYPTED_JWT);
+        MockedStatic<IdGenerator> mockIdGen = Mockito.mockStatic(IdGenerator.class);
+        mockIdGen.when(IdGenerator::generate).thenReturn(TEST_UUID);
+    }
+
+    @Test
+    void shouldReturn302WithEncryptedJWT() throws JOSEException, Json.JsonException {
+        try (MockedConstruction<State> mockedState =
+                Mockito.mockConstruction(
+                        State.class,
+                        (mock, context) -> {
+                            when(mock.getValue()).thenReturn(TEST_STATE_VALUE);
+                        })) {
+            APIGatewayProxyResponseEvent ipvRedirectResponse =
+                    mfaResetIPVAuthorizationService.sendMfaResetRequestToIPV(
+                            TEST_SUBJECT, TEST_CLIENT_SESSION_ID, TEST_SESSION, auditContext);
+            String redirectURI = ipvRedirectResponse.getHeaders().get(ResponseHeaders.LOCATION);
+            RSAPublicKey expectedPublicKey =
+                    new RSAKey.Builder((RSAKey) JWK.parseFromPEMEncodedObjects(TEST_PUBLIC_KEY))
+                            .build()
+                            .toRSAPublicKey();
+            verify(jwtService)
+                    .signJWT(TEST_SIGNING_ALGORITHM, constructTestClaimSet(), TEST_KEY_ALIAS);
+            verify(jwtService).encryptJWT(TEST_SIGNED_JWT, expectedPublicKey);
+            verify(redisConnectionService)
+                    .saveWithExpiry(
+                            TEST_STATE_STORAGE_PREFIX + SESSION_ID,
+                            objectMapper.writeValueAsString(TEST_STATE_VALUE),
+                            TEST_SESSION_EXPIRY);
+            verify(tokenService).generateStorageTokenForMfaReset(TEST_SUBJECT);
+            assertEquals(302, ipvRedirectResponse.getStatusCode());
+            assertEquals("", ipvRedirectResponse.getBody());
+            assertEquals(
+                    TEST_IPV_AUTHORIZE_URI
+                            + "?response_type=code"
+                            + "&request="
+                            + TEST_ENCRYPTED_JWT.serialize()
+                            + "&client_id="
+                            + TEST_IPV_AUTH_CLIENT_ID,
+                    redirectURI);
+            verify(configurationService).getMfaResetJarSigningKeyAlias();
+            verify(configurationService).getStorageTokenClaimName();
+            verify(configurationService).getMfaResetCallbackURI();
+            verify(configurationService).getIPVAuthEncryptionPublicKey();
+            verify(configurationService, times(2)).getIPVAuthorisationClientId();
+            verify(configurationService).getAuthIssuerClaim();
+            verify(configurationService).getIPVAuthorisationURI();
+            verify(configurationService).getSessionExpiry();
+            verify(configurationService).getIPVAudience();
```

Comment:
Mockito say it's best practice to verify every mock interaction like this, but I tend to prefer to assert on the "so what". Like does the produced object contain the audience that I configured the mock to return? It means you have to change your tests less if you change the implementation, and you're testing the behaviour you actually care about

Reply:
As above, have updated the the tests and removed these assertions against the config service